subprojects/core/src/main/java/org/gradle/execution/EntryTaskSelector.java

```diff
@@ -18,10 +18,14 @@
 
 import org.gradle.api.internal.GradleInternal;
 import org.gradle.execution.plan.ExecutionPlan;
+import org.gradle.internal.buildtree.BuildTreeWorkGraph;
 
 public interface EntryTaskSelector {
     void applyTasksTo(Context context, ExecutionPlan plan);
 
+    default void postProcessExecutionPlan(BuildTreeWorkGraph.FinalizedGraph plan) {
```

Comment:
`FinalizedGraph` should be replaced with something that allows the scheduled (but not yet finalized) tasks to be visited.